libs/api/domains/license-service/src/lib/client/adr-license-client/adrLicenseService.api.ts

```diff
@@ -22,73 +33,140 @@ export class GenericAdrLicenseApi implements GenericLicenseClient<AdrDto> {
+        pkpassStatus: GenericUserLicensePkPassStatus.Available,
+      }
+    }
 
     return {
-      status: GenericUserLicenseStatus.HasLicense,
+      status: GenericUserLicenseStatus.NotAvailable,
       payload,
       pkpassStatus: GenericUserLicensePkPassStatus.NotAvailable,
     }
   }
 
   async getLicenseDetail(
     user: User,
+    locale: Locale,
+    labels: GenericLicenseLabels,
   ): Promise<GenericLicenseUserdataExternal | null> {
-    return this.getLicense(user)
+    return this.getLicense(user, locale, labels)
+  }
+
+  private async createPkPassPayload(user: User): Promise<PassDataInput | null> {
+    const license = await this.fetchLicense(user)
+    if (!license) {
+      return null
+    }
+
+    const inputValues = createPkPassDataInput(license)
+    if (!inputValues) return null
+    //Fetch template from api?
+    return {
+      inputFieldValues: inputValues,
+    }
   }
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+
   async getPkPassUrl(user: User): Promise<string | null> {
-    return null
+    const payload = await this.createPkPassPayload(user)
+
+    if (!payload) {
+      return null
+    }
+
+    const pass = await this.smartApi.generatePkPassUrl(
+      payload,
+      format(user.nationalId),
+    )
+    return pass ?? null
   }
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   async getPkPassQRCode(user: User): Promise<string | null> {
-    return null
+    const payload = await this.createPkPassPayload(user)
+
+    if (!payload) {
+      return null
+    }
+    const pass = await this.smartApi.generatePkPassQrCode(
+      payload,
+      format(user.nationalId),
+    )
+
+    return pass ?? null
   }
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   async verifyPkPass(data: string): Promise<PkPassVerification | null> {
-    return null
+    const { code, date } = JSON.parse(data) as PkPassVerificationInputData
+    const result = await this.smartApi.verifyPkPass({ code, date })
+
+    if (!result) {
+      this.logger.warn('Missing pkpass verify from client', {
+        category: LOG_CATEGORY,
+      })
+      return null
+    }
+
+    let error: PkPassVerificationError | undefined
+
+    if (result.error) {
+      let data = ''
+
+      try {
+        data = JSON.stringify(result.error.serviceError?.data)
+      } catch {
+        // noop
+      }
+
+      // Is there a status code from the service?
+      const serviceErrorStatus = result.error.serviceError?.status
+
+      // Use status code, or http status code from serivce, or "0" for unknown
+      const status = serviceErrorStatus ?? (result.error.statusCode || 0)
+
+      error = {
+        status: status.toString(),
+        message: result.error.serviceError?.message || 'Unknown error',
+        data,
+      }
+
+      return {
+        valid: false,
+        data: undefined,
+        error,
+      }
+    }
+
+    /*
+      TODO: VERIFICATION!!!!!!!! Máni (thorkellmani @ github)
```

Comment:
There are so many todo's @thorkellmani 

I wonder if they require a better explanation on why its a TODO and not being resolved in this PR. 🙏

Reply:
improved the todos